src/durableorchestrationclient.ts

```diff
@@ -0,0 +1,443 @@
+     *  the event.
+     * @param connectionName The name of the connection string associated with
+     *  `taskHubName.`
+     * @returns A promise that resolves when the event notification message has
+     *  been enqueued.
+     *
+     * In order to handle the event, the target orchestration instance must be
+     * waiting for an event named `eventName` using
+     * [[waitForExternalEvent]].
+     *
+     * If the specified instance is not found or not running, this operation
+     * will have no effect.
+     */
+    public async raiseEvent(
+        instanceId: string,
+        eventName: string,
+        eventData: unknown,
+        taskHubName?: string,
+        connectionName?: string,
+        ): Promise<void> {
+        if (!eventName) {
+            throw new Error("eventName must be a valid string.");
+        }
+
+        const idPlaceholder = this.clientData.managementUrls.id;
+        let url = this.clientData.managementUrls.sendEventPostUri
+            .replace(idPlaceholder, instanceId)
+            .replace(this.eventNamePlaceholder, eventName);
+
+        if (taskHubName) {
+            url = url.replace(this.clientData.taskHubName, taskHubName);
+        }
+
+        if (connectionName) {
+            url = url.replace(/(connection=)([\w]+)/gi, "$1" + connectionName);
+        }
+
+        const res = await this.webhookClient.post(new URL(url), eventData);
+        switch (res.status) {
+            case 202: // event acccepted
+            case 410: // instance completed or failed
+                return;
+            case 404:
+                throw new Error(`No instance with ID '${instanceId}' found.`);
+            case 400:
+                throw new Error("Only application/json request content is supported");
+            default:
+                throw new Error(`Webhook returned unrecognized status code ${res.status}: ${res.body}`);
+        }
+    }
+
+    /**
+     * Rewinds the specified failed orchestration instance with a reason.
+     * @param instanceId The ID of the orchestration instance to rewind.
+     * @param reason The reason for rewinding the orchestration instance.
+     * @returns A promise that resolves when the rewind message is enqueued.
+     *
+     * This feature is currently in preview.
+     */
+    public async rewind(instanceId: string, reason: string): Promise<void> {
+        const idPlaceholder = this.clientData.managementUrls.id;
+        const url = this.clientData.managementUrls.rewindPostUri
+            .replace(idPlaceholder, instanceId)
+            .replace(this.reasonPlaceholder, reason);
+
+        const res = await this.webhookClient.post(new URL(url));
+        switch (res.status) {
+            case 202:
+                return;
+            case 404:
+                throw new Error(`No instance with ID '${instanceId}' found.`);
+            case 410:
+                throw new Error(`The rewind operation is only supported on failed orchestration instances.`);
```

Comment:
Coming back to this post-release. I agree that it would be best to flow any messages from the extension, but the HTTP responses do not currently contain error messages. We'll have to tweak things in the extension to enable this.